drivers/disk/Kconfig.flash

```diff
@@ -3,6 +3,7 @@
 
 config DISK_DRIVER_FLASH
 	bool "Flash"
+	default y if DT_HAS_ZEPHYR_FLASH_DISK_ENABLED
```

Comment:
This means that in, for example, MCUboot you have to explicitly disable the `CONFIG_DISK_DRIVER_FLASH`, as it will be compiled, whether used or not, just because application that is using the same overlay has the flash disk enabled.

Reply:
Only if `DISK_DRIVERS` are enabled? And if `DISK_DRIVERS` are enabled, then enabling the Flash disk driver seems like the correct behavior

Reply:
Ah, true.